src/frontend/src/scheduler/local.rs

```diff
@@ -82,6 +82,10 @@ impl LocalQueryExecution {
     /// We can convert a query to plan fragment since in local execution mode, there are at most
     /// two layers, e.g. root stage and its optional input stage. If it does have input stage, it
     /// will be embedded in exchange source, so we can always convert a query into a plan fragment.
+    ///
+    /// We remark that the boundary to determine which part should be executed on the frontend and
+    /// which part should be executed on the backend is the first exchange operator when looking
+    /// from the the root of the plan to the leaves.
```

Comment:
The missing part is to embed pushed down plan fragment in exchange info, see `convert_plan_node`

Reply:
do you mean the comments miss this part `embed pushed down plan fragment in exchange info`? I will add it

Our understanding seems the same, i.e. the first exchange operator would contain the pushed-down plan fragment.

Reply:
Right 👌